### DISCUSSION TITLE
Allow model creators to select license terms for their models

### DISCUSSION THREAD

**Discussion by JustMaier** [civitai/civitai#101] [Ideas]
I think it probably makes the most sense to allow toggles that will correspond with Creative Commons (CC) licenses that sit on top of the require openrail license (I'm am not a lawyer and I don't know if this actually makes sense yet).

Dialog and options:

**When using my model, I give permission for users to:**
- Use without crediting me
- Use for commercial purposes
- Make derivatives of this model (Merges)
- Make derivatives with a different license (only if derivatives allowed)

If they leave everything selected then it'd be a CC0 license (aka public domain).

Based on these toggles we could since display the appropriate CC badge and enable/disable running the model on run partners etc.

**Comment by JustMaier** [comment]
**This is live!**
We opted to just display indicators instead of an actual Creative Commons license due to the uncertainty around copyright ownership and AI work.

**Setting model permissions**
![image](https://user-images.githubusercontent.com/607609/210704238-5c1ce32c-e1b7-42b3-adde-a8c04b5f0a4e.png)

**Viewing model permissions**
![image](https://user-images.githubusercontent.com/607609/210704167-06f2ffdd-f118-4538-98ef-cc009531243e.png)

**Comment by 67372a** [comment]
I see that the additional indicators were added, but you have to click them, and they don't standout significantly, users can easily download without ever seeing them.

I'd suggest adding "clickwrap" so they are clearly presented to users and they must click agree.
